calc/vxlan_resolver.go

```diff
@@ -423,25 +490,80 @@ func (c *VXLANResolver) routeReady(r vxlanRoute) bool {
 		logCxt.Debug("Don't yet know the VTEP for this route")
 		return false
 	}
+
+	if _, ok := c.nodeNameToNode[c.hostname]; !ok {
+		logCxt.Debug("No cidr block yet for local node")
+		return false
+	}
+
+	if _, ok := c.nodeNameToNode[r.node]; !ok {
+		logCxt.Debug("No cidr block yet for node")
+		return false
+	}
+
 	return true
 }
 
+func (c *VXLANResolver) nodeCidr(nodeName string) (*cnet.IPNet, error) {
+	logCxt := logrus.WithField("node", nodeName)
+
+	node := c.nodeNameToNode[nodeName]
+	bgp := node.Spec.BGP
+	if bgp == nil {
+		return nil, errors.New("no bgp information for node")
+	}
+
+	_, cidr, err := cnet.ParseCIDROrIP(bgp.IPv4Address)
+	if err != nil {
+		logCxt.Error(err.Error())
+		return nil, err
+	}
+
+	return cidr, nil
+}
+
 // kickPendingRoutes loops through the provided routes to see if there are any which are now programmable,
 // and will send any that are.
 func (c *VXLANResolver) kickPendingRoutes(pendingRouteUpdates set.Set) {
 	pendingRouteUpdates.Iter(func(item interface{}) error {
 		r := item.(vxlanRoute)
-		logCxt := logrus.WithField("route", r)
+		logCxt := logrus.WithField("route", r).WithField("debug", "brian")
 		if !c.routeReady(r) {
 			return nil
 		}
 		logCxt.Info("Sending VXLAN route update")
-		c.callbacks.OnRouteUpdate(&proto.RouteUpdate{
+
+		routeUpdate := &proto.RouteUpdate{
 			Type: proto.RouteType_VXLAN,
 			Node: r.node,
 			Dst:  r.dst.String(),
 			Gw:   c.determineGatewayForRoute(r),
-		})
+		}
+
+		pool := c.vxlanPoolForRoute(r)
```

Comment:
Logically, it feels like this belongs in `determineGatewayForRoute`.

I think it probably makes sense to augment that function with this logic. This probably also helps with the "readiness" checks above. I think it will automatically get us the right readiness logic for cross subnet routes.

Reply:
good point, I'll update it